libs/remote_storage/src/s3_bucket.rs

```diff
@@ -689,10 +689,14 @@ impl RemoteStorage for S3Bucket {
             );
             let versions = response.versions.unwrap_or_default();
             let delete_markers = response.delete_markers.unwrap_or_default();
-            let new_versions = versions.into_iter().map(VerOrDelete::Version);
-            let new_deletes = delete_markers.into_iter().map(VerOrDelete::DeleteMarker);
+            let new_versions = versions.into_iter().map(VerOrDelete::from_version);
+            let new_deletes = delete_markers
```

Comment:
```suggestion
            let versions = versions.into_iter().map(VerOrDelete::from_version);
            let deletes = delete_markers
```

No harm in shadowing? I think it's easier the less names you have. Kind of would like to see these statements swapped, but I didn't yet ...

```
let versions = response.versions
    .unwrap_or_default()
    .into_iter()
    .map(VerOrDelete::from_version);

let deletes = response.delete_markers
    .unwrap_or_default()
    .into_iter()
    .map(VerOrDelete::from_delete_marker);
```

Whoa, there is a lot of `Option`, even around `Vec`. Wonder what the `None` would mean..

Reply:
Done, please check

Reply:
Much nicer, thanks.